### ISSUE TITLE
pdf export does not fire with bigger networks. 

### ISSUE THREAD

**Issue by lukasGeisseler** [cytoscape/cytoscape.js-pdf-export#2]
<img width="416" alt="image" src="https://github.com/user-attachments/assets/4c6affdd-1b0f-450f-8ae6-1d3396ee47c7">. 

I use the package in the above manner (loaded via unpkg).  

btw. I'm new in .js so probably I missed a point.   

Nevertheless, for small graphs your package works fine, but if I change the graph to a larger (about 800 nodes and 1000 vertices), the save pop-up does not appear.  Therefore, I assume that the size matters. Are there other experiences?

**Comment by mikekucera** [comment]
Its possible you are running into some other bug that is preventing the PDF from being generated. Is there an error trace in the console?

**Comment by lukasGeisseler** [comment]
thanks for the fast reply.

that's a good point. 

I have the following error: 

```
cytoscape.umd.js:28797 Uncaught (in promise) TypeError: context.fillRect is not a function
    at CRp$6.drawText (cytoscape.umd.js:28797:21)
    at CRp$6.drawElementText (cytoscape.umd.js:28631:9)
    at drawText (cytoscape.umd.js:28172:9)
    at CRp$8.drawEdge (cytoscape.umd.js:28190:5)
    at CRp$9.drawElement (cytoscape.umd.js:27947:9)
    at CRp$9.drawElements (cytoscape.umd.js:28072:9)
    at createPdfBlob (VM3922 pdf-export.js:209:14)
    at Core.pdfExport [as pdf] (VM3922 pdf-export.js:87:22)
    at cy.on.window.pdf (code.js:35361:40)
    at _loop2 (cytoscape.umd.js:11537:39)
```

Probably i did something wrong. Nevertheless, if I change the graph data to the one provided by the example, it works fine.

**Comment by mikekucera** [comment]
You're not doing anything wrong. This extension is very new and hasn't been tested with many real-world networks yet. I'm expecting bugs like this to come up.

The actual issue is that a call to fillRect(...) in cytoscape is not being handled correctly. I should be able to fix this.

In the meantime if you want to work around this... it appears the call to fillRect() is happening because some labels have a background. If you are using the 'text-background-color' or 'text-background-opacity' styles try disabling them for now.

**Comment by mikekucera** [comment]
Fixed and published 0.0.3 to npm

**Comment by lukasGeisseler** [comment]
Wow! You are fast. Thanks a lot!
